core/src/main/scala/com/mesosphere/usi/core/SchedulerLogicHandler.scala

```diff
@@ -85,13 +86,36 @@ private[core] class SchedulerLogicHandler(mesosCallFactory: MesosCalls) {
     */
   private var cachedPendingLaunch = CachedPendingLaunch(Set.empty)
 
+  def validateEvent(msg: SpecEvent): Seq[PodInvalid] = msg match {
+    case SpecsSnapshot(podSpecs, _) =>
+      podSpecs
+        .map(p => {
+          PodSpec.isValid(p) match {
+            case Nil => None
+            case err => Some(PodInvalid(p.id, err))
+          }
+        })
+        .collect { case Some(p) => p }
+    case PodSpecUpdated(_, Some(podSpec)) =>
+      PodSpec.isValid(podSpec) match {
```

Comment:
another neat trick is to use `toList` method:
```
@ Some(1).toList
res2: List[Int] = List(1)

@ None.toList
res3: List[Nothing] = List()
```
so simply:
```suggestion
      PodSpec.isValid(podSpec).toList
```